### PR TITLE
[WIP]Request access to join private channels.

### PR DESCRIPTION
**Pull request by vickyokrm**
1) Request access to join the private room
2) Adapt the new directory search to display `Private Groups`
3) Introduce `Secrecy` to the private channels.

### CODE REVIEW

File: packages/rocketchat-ui/client/views/app/createChannel.js [hunk 1]
```diff
@@ -164,6 +170,9 @@ Template.createChannel.events({
 	},
 	'change [name="type"]'(e, t) {
 		t.type.set(e.target.checked ? e.target.value : 'd');
+		if (!e.target.check) {
```
Comment (mrsimpson) [inline packages/rocketchat-ui/client/views/app/createChannel.js]: How does this reflect the comment? I don't see a validation with respect to public channels only. And afterall: should this not be `checked`?

File: tests/chimp-config.js [hunk 2]
```diff
@@ -19,7 +19,7 @@ module.exports = {
 // 	showXolvioMessages: true,
 
 // 	// - - - - CUCUMBER - - - -
-	path: 'tests/end-to-end',
+	path: 'tests/end-to-end/ui_dummy',
```
Comment (mrsimpson) [inline tests/chimp-config.js]: ?

**Comment by ruKurz** [comment]
- [ ] If a user is removed actively from a private channel he is not able to join it again (Button not displayed)
- [ ] channel content that was written before the request has been approved, is not showing before the requester is refreshing the browser
- [ ] use the lock symbol in the discovery dialog for private channels
- [ ] The wording for visibility should be "versteckt" instead "geheim"
- [ ] Notifications are ending on English verbs on "pending/declined/...", German translation needed
- [ ] Check German translations (e.g. dialog title label to join a private channel should be "Kanal beitreten")

**Comment by mrsimpson** [comment]
@vickyokrm with 0.69, there is a new API in RC which renders buttons:
https://github.com/RocketChat/Rocket.Chat/pull/11473

File: packages/rocketchat-lib/client/lib/openRoom.js [hunk 3]
```diff
@@ -31,17 +31,23 @@ function openRoom(type, name) {
...
-							return BlazeLayout.render('main', {center: 'roomNotFound'});
+							if (type === 'p' && err.error === 'error-no-permission') {
+								Session.set('privateNoPermission', { type, name });
+								return BlazeLayout.render('main', { center: 'privateNoPermission' });
+							} else {
+								Session.set('roomNotFound', { type, name });
+								return BlazeLayout.render('main', { center: 'roomNotFound' });
+							}
 						} else {
+							delete record.$loki; // TODO: Why loki used here
```
Comment (mrsimpson) [inline packages/rocketchat-lib/client/lib/openRoom.js]: why did you remove it? Did that cause issues?

**Comment by ruKurz** [comment]
What is keeping us away merging this PR?

**Comment by mrsimpson** [comment]
Unfortunately, we didn't push this to a core PR. We'll not enlarge our diff, so if we should need this again, we'll keep this closed PR as reference
